portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -115,23 +114,40 @@ private String getParameterOrDefault(HttpServletRequest request, String name, St
 
   /*
    * Return a strictly positive integer from a url parameter, or defaultValue if not present
-   * Note: assumes defaultValue is strictly positive
+   * If defaultValue and the requested value are not positive, 1 will be returned. 
    */
-  private int getPositiveIntInputOrDefault(HttpServletRequest request, String name, String defaultValue) {
+  private int getPositiveInputOrDefault(HttpServletRequest request, String name, int defaultValue) {
     int value = getIntInputOrDefault(request, name, defaultValue);
-    return value > 0 ? value : Integer.parseInt(defaultValue);
+    if (value < 1) {
+      //additional guarantee that defaultValue is also positive...
+      return defaultValue > 0 ? defaultValue : 1;
+    }
+    return value;
   }
 
   /*
    * Return an integer in the url parameter, or defaultValue if not present
    */ 
-  private int getIntInputOrDefault(HttpServletRequest request, String name, String defaultValue) {
+  private int getIntInputOrDefault(HttpServletRequest request, String name, int defaultValue) {
     try {
-      int inputValue = Integer.parseInt(getParameterOrDefault(request, name, defaultValue));
+      int inputValue = Integer.parseInt(getParameterOrDefault(request, name, 
```

Comment:
This code is more complex than just inlining getParameterOrDefault.

Reply:
Will rewrite in a future branch